Throw if container is missing fragments

devknoll opened the pull request:
Another crack at #20 / #131

Rather than building the query set from the set of all fragments that a component has, it builds them from the set **required** by the route, and throws if the component doesn't provide one.

I'm on the fence about keeping `hasFragment`.

This should be compatible with FB code, but let me know if it's not @steveluscher @josephsavona

josephsavona commented on src/container/getRelayQueries.js:
```diff
@@ -46,9 +46,16 @@ function getRelayQueries(
+      route.name,
+      Component.displayName,
+      fragmentName
+    );
```
Let's keep the message consistent with the below:

```javascript
'Relay.QL defined on route `%s` named `%s` is not a valid query, ' +
'a matching fragment was not found on component `%s`.',
route.name,
fragmentName,
Component.displayName
```

josephsavona replied:
GH won't let me add a comment to a non-updated line, so can you also change line 68-69 to:

```javascript
invariant(
  ...,
  'Relay.QL: query `%s.queries.%s` is invalid, a typical query is defined using: ' +
  '() => Relay.QL`query { ... }`.',
  route.name,
  queryName
```

josephsavona commented on src/container/__tests__/getRelayQueries-test.js:
```diff
@@ -122,4 +123,22 @@ describe('getRelayQueries', () => {
     );
   });
+
+  it('throws if a container doesnt include a required fragment', () => {
```
s/doesnt/does not/ (grammar)

josephsavona commented on src/container/getRelayQueries.js:
```diff
@@ -46,9 +46,16 @@ function getRelayQueries(
-    // TODO: Fix this. It relies on the query and fragment names matching.
-    var queryName = fragmentName;
+  Object.keys(route.queries).forEach(queryName => {
+    var fragmentName = queryName;
```
just use `queryName`

devknoll commented:
Fixed. @josephsavona do you have any thoughts about `hasFragment`? I don't really like bloating the RelayContainer interface...

josephsavona commented on src/container/__tests__/RelayContainer-test.js:
```diff
@@ -210,6 +210,12 @@ describe('RelayContainer', function() {
       ));
     });
+
+    it('can tell if a fragment was given', () => {
```
`it('returns whether a named fragment is defined', ...`

josephsavona commented:
@devknoll looks like you made updates before i finished adding all my comments. Can you double-check all of the feedback?

`hasFragment()` seems reasonable.

devknoll commented:
:white_check_mark: Updated missing fragment invariant
:white_check_mark: Updated invalid query invariant
:white_check_mark: Changed to 'does not'
:white_check_mark: Nixed fragmentName
:white_check_mark: Fixed test name

josephsavona commented:
@devknoll we're basically all set to merge this once the merge conflicts are resolved

devknoll commented:
Whoops, thanks for the heads up @josephsavona

josephsavona commented:
@devknoll awesome job on this.
